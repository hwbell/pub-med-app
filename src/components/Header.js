import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

var ncbi = require('node-ncbi');

// elements

// styling
// import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

// elements

class Header extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className={this.props.class || "glass"}>
        
          <h2 className="title">{this.props.title}</h2>
          <h4 className="subtitle">{this.props.subtitle}</h4>

      </div>
    );
  }
}

const styles = {
  main: {
    padding: '10px'
  },
  text: {
    color: 'white'
  }

}

export default Header;
