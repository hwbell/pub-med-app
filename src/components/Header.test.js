import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', async () => {
  let wrapper = mount(<Header />);
});

it('renders correctly', async () => {
  const tree = await renderer
    .create(<Header/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the correct elements', () => {
  let wrapper = shallow( <Header/>);
  expect(wrapper.find('.glass').length).toEqual(1);
  expect(wrapper.find('.title').length).toEqual(1);
  expect(wrapper.find('.subtitle').length).toEqual(1);

})