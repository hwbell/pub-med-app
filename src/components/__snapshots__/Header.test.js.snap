// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col glass"
>
  <h2
    className="title"
  />
  <h4
    className="subtitle"
  />
</div>
`;
